content/rsf-spec/index.md

```diff
@@ -0,0 +1,263 @@
+```
+
+#### <a id="append-entry-command">`append-entry` command</a>
+
+Appends a new [Entry resource][entry-res] to the register.
+
+##### Arguments
+
+1. The `type` of the entry determines if the entry belongs to the data log
+   (`user`) or to the metadata log (`system`).
+2. The `key` of the entry. The primary key field is the field with the same
+   name as the register.
+3. The `timestamp` of the entry. This is the time at which the entry was
+   appended to the register.
+4. The `hash` of the item for which the entry was appended. This is the
+   [sha-256 hash of the item][canon-rep].
+
+For example:
+
+```
+append-entry	user	GB	2010-11-12T13:14:15Z	sha-256:08bef0039a4f0fb52f3a5ce4b97d7927bf159bc254b8881c45d95945617237f6
+```
+
+
+### Validation rules
+
+A RSF list of commands is expected to conform to the following rules:
+
+* [Commands](#commands) are executed in order of appearance, top to bottom.
+* Entries are numbered in sequence in order of appearance starting with 1 if
+  the register is empty, otherwise incrementing on the latest entry number
+  found in the register.
+* An [`append-entry` command](#append-entry-command) must always appear after
+  the [`add-item` command](#add-item-command) that introduces the item is
+  referencing.
+* It is illegal to have orphan items. An `add-item` must have at least one
+  `append-entry` referencing to the item.
+* It is illegal to have broken references. An `append-entry` must reference an
+  item previously introduced by an `add-item` command.
+* The item in the `add-item` command must always be in the canonical form.
+
+
+#### Type checking
+
+Although not part of the RSF specification, it is worth mentioning that a
+Registers' implementation is expected to type check the data according to the
+computed schema.
+
+##### Metadata
+
+A metadata item must conform to the metadata schema
+
+[TODO: This needs definition].
+
+##### Data
+
+A data item must conform to the current schema derived from the previous
+system entries. A type checker is expected to verify:
+
+* It has the primary key defined.
+* Fieldnames exist in the schema.
+* Cardinality is consistent.
+* Datatype is consistent.
+
+Given the example “[All commands in use](#all-commands-example)”, a new data
+item is valid if:
+
+* It has the primary key, `country` defined.
+* It has at most one `name` field and one `official-name` field.
+* The `country` field has cardinality 1.
+* The `country` field is a String.
+* The `name` field has cardinality 1.
+* The `name` field is a String.
+* The `official-name` field has cardinality 1.
+* The `official-name` field is a String.
+
+Each datatype must be parsed according to the [datatype specification][datatype-spec].
+
+A RSF patch (set of commands) must be treated as a single transaction. If
+there is a validation error, the whole patch must be rejected and any changes
+to the state rolled back.
+
+
+### Examples
```

Comment:
Should these examples be valid patches?

Reply:
They should yes, if I haven't mess it up, they are 😱

Reply:
Haha, I don't think you've messed them up.  But I think the spec should require an `assert-root-hash` in the first and final lines.